### PR TITLE
feat(storage-apps): add dell-csi-unity

### PR DESCRIPTION
**Pull request by vmaillot**
Signed-off-by: Valentin Maillot <valentin.maillot@adfinis.com>

# Description

Add Dell CSI Unity chart into `storage-apps` as we need it for [Dectris](https://github.com/dectris/argocd-deployments/issues/143).

# Issues

[#143](https://github.com/dectris/argocd-deployments/issues/143)

# Checklist

* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run -a`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] If I updated a dependency tool, or app, this PR contains a short summary of the changes I'm pulling
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

**Review by eyenx — APPROVED** [review]
420!

**Review by eyenx — CHANGES_REQUESTED** [review]
Stahp

File: charts/storage-apps/values.yaml [hunk 1]
```diff
@@ -57,3 +57,23 @@ cephCsiCephfs:
...
+  name: dell-csi-unity
+  destination:
+    # dellCsiUnity.destination.namespace -- Namespace
+    namespace: "infra-storage"
+  # dellCsiUnity.repoURL -- Repo URL
+  # @default -- [repo](https://github.com/dell/csi-unity/tree/main/helm/csi-unity)
+  repoURL: "https://github.com/dell/csi-unity/tree/main/helm"
+  # dellCsiUnity.chart -- Chart
+  path: "csi-unity"
+  # dellCsiUnity.targetRevision -- [ceph-csi-cephfs Helm chart](https://github.com/ceph/csi-charts/tree/master/docs/cephfs) version
```
Comment (eyenx) [inline charts/storage-apps/values.yaml:75]: Wrong url

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/storage-apps/values.yaml [hunk 2]
```diff
@@ -57,3 +57,23 @@ cephCsiCephfs:
...
+  targetRevision: "2.0.*"
+  # dellCsiUnity.values -- Helm values
+  # @default -- [upstream values](https://github.com/ceph/csi-charts/tree/master/docs/cephfs/ceph-csi-cephfs/values.yaml)
```
Comment (eyenx) [inline charts/storage-apps/values.yaml:78]: Wrong url

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/storage-apps/README.md [hunk 3]
```diff
@@ -37,6 +37,12 @@ This chart is maintained by [Adfinis](https://adfinis.com/?pk_campaign=github&pk
 | cephCsiRbd.repoURL | string | [repo](https://ceph.github.io/csi-charts) | Repo URL |
 | cephCsiRbd.targetRevision | string | `"3.4.*"` | [ceph-csi-rbd Helm chart](https://github.com/ceph/csi-charts/tree/master/docs/rbd) version |
 | cephCsiRbd.values | object | [upstream values](https://github.com/ceph/csi-charts/tree/master/docs/rbd/ceph-csi-rbd/values.yaml) | Helm values |
+| dellCsiUnity | object | - | [dell-csi-unity](https://github.com/dell/csi-unity) |
+| dellCsiUnity.destination.namespace | string | `"infra-storage"` | Namespace |
+| dellCsiUnity.enabled | bool | `false` | Enable Dell CSI-Unity |
+| dellCsiUnity.repoURL | string | [repo](https://github.com/dell/csi-unity/tree/main/helm/csi-unity) | Repo URL |
+| dellCsiUnity.targetRevision | string | `"2.0.*"` | [ceph-csi-cephfs Helm chart](https://github.com/ceph/csi-charts/tree/master/docs/cephfs) version |
```
Comment (eyenx) [inline charts/storage-apps/README.md:44]: Wrong url

File: charts/storage-apps/README.md [hunk 4]
```diff
@@ -37,6 +37,12 @@ This chart is maintained by [Adfinis](https://adfinis.com/?pk_campaign=github&pk
...
+| dellCsiUnity.values | object | [upstream values](https://github.com/ceph/csi-charts/tree/master/docs/cephfs/ceph-csi-cephfs/values.yaml) | Helm values |
```
Comment (eyenx) [inline charts/storage-apps/README.md:45]: Wrong url

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/storage-apps/values.yaml (hunk 1 above)
Comment (vmaillot) [reply charts/storage-apps/values.yaml:75]: but there's no chart repo for this? can't we use a github repo?

**Comment by vmaillot** [comment]
> Stahp

hm?

**Review by eyenx — APPROVED** [review]

File: charts/storage-apps/values.yaml (hunk 1 above)
Comment (hairmare) [reply charts/storage-apps/values.yaml]: Without a chart repo, how does versioning (pinning and wildcarding) work :confused: :thinking:

**Review by hairmare — CHANGES_REQUESTED** [review]
I think this needs more use-cases before we add "generic" support in here

File: charts/storage-apps/values.yaml (hunk 1 above)
Comment (vmaillot) [reply charts/storage-apps/values.yaml]: > Without a chart repo, how does versioning (pinning and wildcarding) work confused thinking

that's why i was wondering... is that even possible?

**Comment by vmaillot** [comment]
as we have only one use case for this chart, i close this pr.

**Pull request closed without merging**